client/components/common/BaseForm.vue

```diff
@@ -0,0 +1,111 @@
+<!-- Reusable component representing a form that can be used in multiple contexts -->
+
+<template>
+  <article>
+    <span
+      v-for="{id, label, type, options, hint, optional} in fields"
+      :key="id"
+    >
+      <label v-if="type !== 'hidden'" :for="id"> {{ label }}: </label>
+      <small v-if="optional"> (optional) </small>
+      <select
+        v-if="type === 'select'"
+        :class="errors[id] ? 'error' : ''"
+        v-model="values[id]"
+        @change="() => validate(id)"
+      >
+        <option/>
+        <option
+          v-for="option in options"
+          :key="option"
+          :value="option"
+        >
+        {{ option }}
+        </option>
+      </select>
+      <textarea
+        v-else-if="type === 'textarea'"
+        :class="errors[id] ? 'error' : ''"
+        v-model="values[id]"
+        @change="() => validate(id)"
+      />
+      <input
+        v-else
+        :class="errors[id] ? 'error' : ''"
+        :type="type || 'text'"
+        v-model="values[id]"
+        @change="() => validate(id)"
+      >
```

Comment:
I wonder if we should incorporate min/nax values for number inputs? Presumably introducing yet another `v-if` here and adds more things into `fields`

Reply:
yeah I thought about doing something like that but I felt like maybe we can just accomplish with a validator? For the scale field we could also go the route of a dropdown or I could add like radio buttons or something if you'd prefer that.